test/Streamly/Test/Internal/Data/Parser.hs

```diff
@@ -0,0 +1,305 @@
+--         Right _ -> False
+--         Left _ -> True)
+--     .&&.
+--     property (case S.parse (P.splitWith (,) (P.yield (1 :: Int)) (P.die "die")) (S.fromList [1 :: Int]) of
+--         Right _ -> False
+--         Left _ -> True)
+--     .&&.
+--     property (case S.parse (P.splitWith (,) (P.die "die") (P.die "die")) (S.fromList [1 :: Int]) of
+--         Right _ -> False
+--         Left _ -> True)
+
+-- teeWith :: Property
+-- teeWith = 
+--     forAll (chooseInt (0, 10000)) $ \n ->
+--         forAll (listOf (chooseInt (0, 1))) $ \ls ->
+--             let
+--                 prsr = P.take n FL.toList
+--             in
+--                 case S.parse (P.teeWith (,) prsr prsr) (S.fromList ls) of
+--                     Right (ls_1, ls_2) -> (Prelude.take n ls == ls_1) && (ls_1 == ls_2)
+--                     Left _ -> False
+--     .&&.
+--     property (case S.parse (P.teeWith (,) (P.die "die") (P.yield (1 :: Int))) (S.fromList [1 :: Int]) of
+--         Right _ -> False
+--         Left _ -> True)
+--     .&&.
+--     property (case S.parse (P.teeWith (,) (P.yield (1 :: Int)) (P.die "die")) (S.fromList [1 :: Int]) of
+--         Right _ -> False
+--         Left _ -> True)
+--     .&&.
+--     property (case S.parse (P.teeWith (,) (P.die "die") (P.die "die")) (S.fromList [1 :: Int]) of
+--         Right _ -> False
+--         Left _ -> True)
+
+-- deintercalate :: Property
+-- deintercalate = 
+--     forAll (listOf (chooseInt (0, 1))) $ \ls ->
+--         case S.parse (P.deintercalate concatFold prsr_1 concatFold prsr_2) (S.fromList ls) of
+--             Right parsed_list_tuple -> parsed_list_tuple == partition (== 0) ls
+--             Left _ -> False
+
+--         where
+--             prsr_1 = (P.takeWhile (== 0) FL.toList)
+--             prsr_2 = (P.takeWhile (== 1) FL.toList)
+--             concatFold = FL.Fold (\concatList curr_list -> return $ concatList ++ curr_list) (return []) return
+
+-- shortest :: Property
+-- shortest =
+--     forAll (listOf (chooseInt(0, 10000))) $ \ls ->
+--         let
+--             prsr_1 = P.takeWhile (<= 2500) FL.toList
+--             prsr_2 = P.takeWhile (<= 5000) FL.toList
+--             prsr_shortest = P.shortest prsr_1 prsr_2
+--         in
+--             case S.parse prsr_shortest (S.fromList ls) of
+--                 Right short_list -> short_list == Prelude.takeWhile (<= 2500) ls
+--                 Left _ -> False
+--     .&&.
+--     property (case S.parse (P.shortest (P.die "die") (P.yield (1 :: Int))) (S.fromList [1 :: Int]) of
+--         Right r -> r == 1
+--         Left _ -> False)
+--     .&&.
+--     property (case S.parse (P.shortest (P.yield (1 :: Int)) (P.die "die")) (S.fromList [1 :: Int]) of
+--         Right r -> r == 1
+--         Left _ -> False)
+--     .&&.
+--     property (case S.parse (P.shortest (P.die "die") (P.die "die")) (S.fromList [1 :: Int]) of
+--         Right _ -> False
+--         Left _ -> True)
+
+main :: IO ()
+main = hspec $ do
```

Comment:
Do write the description in line with how other test files have it. For example, take a look at other  test files `test/Streamly/Test/Array.hs`